web/src/topic_list_data.js

```diff
@@ -62,6 +67,10 @@ export function get_list_info(stream_id, zoomed, search_term) {
                     return false;
                 }
 
+                if (is_topic_unmuted) {
+                    return true;
+                }
```

Comment:
This algorithm is wrong. First, unmuted topics should have no special priority in a stream that is not muted, and this code does not do that. Further, if you have 25 unmuted topics in a stream, this will show all of them in the !zoomed case, since this check is before the `idx < max_topics` one. I don't see an easy way to fix that with this single pass iteration through the topics algorithm like what you have here, because the (inherit) topics have no way to know if we'll later find 10 (old!) unmuted topics, and thus the 6th oldest topic should have been skipped for that reason.

I think the correct thing to do is turn this `for` loop into a function (`choose_topics`) that accepts the list of topics as a parameter. For a stream that is not muted, we just call the function with `topics_names`. For a muted stream, we call that function twice -- first with just the unmuted topics, and then with all the other topics, and then concatenate the results.

It'll need some careful preparatory refactoring; the `idx < max_topics` check probably needs to be `idx < max_topics && topics_selected < max_topics`, since otherwise we will overselect in the second pass if there are several unmuted topics older than the 8th oldest topic.

We'll also need to be careful to ensure the `topics_selected` value is shared between the two loop runs; basically any unmuted topics we've selected should take away from how many (inherit) topics in the muted stream we display after clicking into the stream.